alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/sharding/SolrShardingMetricsContainer.java

```diff
@@ -45,18 +45,20 @@ public void refresh() {
     }
 
     public Set<ShardInstance> getShardInstances(Shard shard) {
-        return rawData.get(shard.getFloc()).get(shard).stream().map(shardState -> shardState.getShardInstance())
+        return rawData.get(shard.getFloc()).get(shard).stream().map(shardState -> (shardState != null ? shardState.getShardInstance():null)).filter(instance -> instance != null)
```

Comment:
don't use that `null` check in the `map()` function, but `filter()` it out first ?

Reply:
```
.stream().filter(shardState -> Objects::nonNull(shardState)).map(shardState ...
```

Reply:
Woops.Forgot about this. Thanks!